rabbitmq/src/test/java/io/micronaut/rabbitmq/intercept/ConsumerRecoveryTest.java

```diff
@@ -0,0 +1,227 @@
+package io.micronaut.rabbitmq.intercept;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.github.dockerjava.api.DockerClient;
+import com.rabbitmq.client.AMQP;
+import com.rabbitmq.client.AlreadyClosedException;
+import com.rabbitmq.client.ConnectionFactory;
+import io.micronaut.context.annotation.Property;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.messaging.annotation.MessageBody;
+import io.micronaut.rabbitmq.annotation.Queue;
+import io.micronaut.rabbitmq.annotation.RabbitListener;
+import io.micronaut.rabbitmq.exception.RabbitListenerException;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import jakarta.inject.Inject;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.containers.GenericContainer;
+
+import static java.lang.Math.abs;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@MicronautTest(rebuildContext = true, propertySources = "classpath:rabbit/recovery-test.yaml")
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@Property(name = "node-name", value = EnvironmentLoader.NODE1)
+public class ConsumerRecoveryTest {
```

Comment:
I would prefer not to bring in Junit for a single test. Is there a technical reason you included Junit or was it just preference?

Reply:
It is preference. I did not use Spock before.